src/librustc/mir/mod.rs

```diff
@@ -1908,24 +1912,24 @@ impl<'tcx> Place<'tcx> {
     //
     // FIXME: can we safely swap the semantics of `fn base_local` below in here instead?
     pub fn local_or_deref_local(&self) -> Option<Local> {
-        match self {
-            Place {
-                base: PlaceBase::Local(local),
-                projection: box [],
+        match self.as_ref() {
```

Comment:
most of these could be pulled out into a separate PR, making the diff here smaller

Reply:
Done